server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -376,9 +321,6 @@ private OnDiskState loadOnDiskState(Path dataPath, DirectoryReader reader) throw
             final Metadata metadata = Metadata.Builder.fromXContent(XContentFactory.xContent(XContentType.SMILE)
                 .createParser(namedXContentRegistry, LoggingDeprecationHandler.INSTANCE, bytes.bytes, bytes.offset, bytes.length));
             logger.trace("found global metadata with last-accepted term [{}]", metadata.coordinationMetadata().term());
-            if (builderReference.get() != null) {
```

Comment:
This change seems unrelated, it's here to catch a case where someone meddled with the metadata and left us with two copies of the global metadata in the same data path. I think we should keep this.